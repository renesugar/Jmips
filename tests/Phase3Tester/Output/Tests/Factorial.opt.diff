--- SelfTestCases/Factorial.opt.out	2019-02-02 14:36:13.000000000 -0800
+++ ./Output/Tests/Factorial.opt.out	2019-03-11 00:03:07.000000000 -0700
@@ -1 +1 @@
-3628800
+0
