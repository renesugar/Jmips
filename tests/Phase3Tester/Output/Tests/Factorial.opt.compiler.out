const empty_Fac

func Main [in 0, out 0, local 0]
  $a0 = :empty_Fac
  $a1 = 10
  call :Fac.ComputeFac
  $t8 = $v0
  PrintIntS($t8)
  ret

func Fac.ComputeFac [in 0, out 0, local 0]
  $t4 = $a0
  $t7 = $a1
  $t8 = LtS($t7 1)
  if0 $t8 goto :if1_else
  $t8 = 1
  goto :if1_end
if1_else:
  $t6 = Sub($t7 1)
  $a0 = $t4
  $a1 = $t6
  call :Fac.ComputeFac
  $t5 = $v0
  $t8 = MulS($t7 $t5)
if1_end:
  $v0 = $t8
  ret


